docs/guides/integration-mobile-client-server.md

```diff
@@ -0,0 +1,374 @@
+:::
+
+## Optional: Reduce Latency by Setting the API Base URL for the Production Environment
+
+<ReduceLatency />
+
+To specify a different UID2 server, you can change it in the `init` call:
+
+<Tabs groupId="language-selection">
+<TabItem value='android' label='Android'>
+
+```js
+UID2Manager.init(
+  context = this,
+  serverUrl = " https://global.prod.uidapi.com"
+)
+```
+
+</TabItem>
+<TabItem value='ios' label='iOS'>
+
+```js
+UID2Settings.shared.environment = .singapore
+// or
+UID2Settings.shared.environment = .custom(
+  url: URL(string: "https://global.prod.uidapi.com")!
+)
+```
+
+</TabItem>
+</Tabs>
+
+## Configure the UID2 Mobile SDK for Your Mobile App
+
+After you've instantiated `UID2Manager` correctly in your mobile app, you'll need to pass a UID2 <Link href="../ref-info/glossary-uid#gl-identity">identity</Link> generated server-side (see [Implement server-side token generation](#implement-server-side-token-generation)), and then pass it into the mobile app using the `setIdentity` method, as shown in the following:
+
+<Tabs groupId="language-selection">
+<TabItem value='android' label='Android'>
+
+```js
+UID2Manager.getInstance().generateIdentity()
+```
+
+</TabItem>
+<TabItem value='ios' label='iOS'>
+
+```js
+UID2Manager.shared.setIdentity()
+```
+
+</TabItem>
+</Tabs>
+
+## Token Storage
+
+After you call the `setIdentity` method, the UID2 Identity is persisted in local file storage.
+
+:::warning
+The format of the file stored in the local file storage, or the filename itself, could change without notice. We recommend that you do not read and update the file directly.
+:::
+
+
+## Pass Generated Token for Bid Stream Use
+
+To retrieve the token, in your mobile app, call the following:
+
+<Tabs groupId="language-selection">
+<TabItem value='android' label='Android'>
+
+```js
+UID2Manager.getInstance().getAdvertisingToken()
+```
+
+</TabItem>
+<TabItem value='ios' label='iOS'>
+
+```js
+UID2Manager.shared.getAdvertisingToken()
+```
+
+</TabItem>
+</Tabs>
+
+If a successful identity was added into the UID2Manager, this method returns a string such as the following:
+
+```
+AgAAAQFt3aNLXKXEyWS8Tpezcymk1Acv3n+ClOHLdAgqR0kt0Y+pQWSOVaW0tsKZI4FOv9K/rZH9+c4lpm2DBpmFJqjdF6FAaAzva5vxDIX/67UOspsYtiwxH73zU7Fj8PhVf1JcpsxUHRHzuk3vHF+ODrM13A8NAVlO1p0Wkb+cccIIhQ==
+```
+
+You can use this identity to pass downstream for sending in the RTB bid stream.
+
+If the `getAdvertisingToken()` method call returns `null`, there was no identity or valid token generated.
+
+One possible reason for this is that the advertising token inside the UID2 identity has expired, and the refresh token has also expired so the SDK cannot refresh the token. If this happens, you'll need to call the `generateIdentity()` method again: see [Implement Server-Side Token Generation](#implement-server-side-token-generation). Then, pass the result into the mobile app’s UID2Manager again.
```

Comment:
@sunnywu your quote was from a different part of the Word doc, "Best Practice on When to Pass New UID2 Token to the UID2 SDK ". This piece is in section "Check that the token was successfully generated and pass it for bid stream use" in the Word doc, which is here as "Pass Generated Token for Bid Stream Use" and the Word doc says:
---------------------------------------------------------------------
You can use this identity to pass downstream for sending in RTB bid stream. 
If  
For Android: UID2Manager.getInstance().getAdvertisingToken() 
For iOS: UID2Manager.shared.getAdvertisingToken() 
returns null, there was not an identity or valid token generated for several reasons such as: 
The identity is invalid 
The advertising token inside the UID2 identity has expired and the refresh token expires too so token cannot be refreshed  
In such situation, you will need to follow Implement Server-Side Token Generation on your backend server section above and generate a new UID2 identity and pass the result into the mobile application’s UID2Manager again. 
---------------------------------------------------------------------
Here's what I updated this to just now (not checked in yet):
-----------------------
If the `getAdvertisingToken()` method call returns `null`, there was no identity or valid token generated. This could be for several reasons, such as:

- The identity is invalid.
- The advertising token inside the UID2 identity has expired, and the refresh token has also expired, so the SDK cannot refresh the token.

If this happens, follow the instructions in [Implement Server-Side Token Generation](#implement-server-side-token-generation) again, generate a new identity, and pass the result into your mobile app's UID2Manager again.
-----------------------